Js2

Snurre1 opened the pull request.

Snurre1 committed:
register and login done

Snurre1 committed:
create/delete/update/getpost JS

Snurre1 committed:
create done

Snurre1 committed:
update working

Snurre1 committed:
update profile done

Snurre1 committed:
Update index.mjs

Snurre1 commented:
done with most of the requirements, need to do the delete post, and filter/search content

Snurre1 committed:
deleted done

not the best of UX but but

Snurre1 committed:
search by ID is working

need to find out how to get more results than 100

Snurre1 committed:
search by id working

need more results though

martink-rsa commented:
Hi Thomas,

Good job so far with the assignment. There is still some work that has to be done, but good job so far.

✔User with @noroff.no or @stud.noroff.no email can register profile
✔Registered user can login
✔User can view a post content feed
❌User can filter the post content feed
❌User can search the post content feed
❌User can view a post content item by ID
✔User can create a post content item
✔User can update a post content item
❌User can delete a post content item
 
## Filtering:
There isn’t something that is filtering or sorting the posts as per the assignment requirements.
## Searching of posts:
The searching of posts does not appear to be working and I don’t see any code relating to searching.
## Can’t view individual post:
There isn’t a way to view an individual post.
## Deleting a post doesn’t working:
Deleting a post does not work.
## Unable to install dependencies:
Bootstrap is set for major package version 2 which has issues when trying to install:
```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for bootstrap@^2.0.0.
npm ERR! notarget In most cases you or one of your dependencies are 
requesting
npm ERR! notarget a package version that doesn't exist.
```
I bumped this to version 5 so I could continue with the review. Please let me know what you’re using to run the project.

## JSDocs:

I didn’t see any JSDocs added to your code which is part of the assignment and JS2 curriculum. Please add JSDocs to some of your functions to demonstrate that you know how to do so.

## Redirect user after login:

After you log in you should consider redirecting the user to the main page or the profile page.

## Edit and delete buttons appear for all posts:

You should only have the edit and delete buttons appearing for posts the user made. This isn’t a requirement however it would be a better user experience.

## Should reload data after a post is made

You should consider reloading the data after a post is made as well as displaying a success message.

Please don’t hesitate to contact the tutors if you need any assistance with the coursework.

Kind regards,
Martin

Snurre1 committed:
filter working

Snurre1 committed:
last push for JS2 CA

some JS docs were made, and checking that I havent left any console.logs up and running

Snurre1 commented:
redirect user after login, check.
some JSdocs added, check.
.filter() added, check.
delete post works on my end, so dont know whats wrong there.
searching post by ID is working, check.
my bootstrap depencdencies says : "^2.0.0" had some issues with it when I had to redo the frameworks CA.
I`m only grabbing 100 results, so be aware of that when search post by ID:D 
Gotta learn how to grab more results.
Edit and delete button, didnt get a chance to change that, didnt really know how to go forward on that one.
